src/linearcombination.jl

```diff
@@ -60,6 +62,67 @@ function A_mul_B!(y::AbstractVector, A::LinearCombination, x::AbstractVector)
     return y
 end
 
+else # 5-arg mul! is available for matrices
+
+# map types that have an allocation-free 5-arg mul! implementation
+const FreeMap = Union{MatrixMap,UniformScalingMap}
+
+function A_mul_B!(y::AbstractVector, A::LinearCombination, x::AbstractVector)
+    # no size checking, will be done by individual maps
+    A_mul_B!(y, A.maps[1], x)
+    l = length(A.maps)
+    if l>1
+        z = similar(y)
+        for n in 2:l
+            An = A.maps[n]
+            if An isa FreeMap
+                mul!(y, An, x, true, true)
+            else
```

Comment:
One trick I learned recently is to add, in the body of a loop, a statement like
```julia
if !(@isdefined z)
     z = similar(y)
end
```
which would then only allocate `z` once the first map which is not a `FreeMap` is encountered, and thus not allocate at all if all the maps are `FreeMap`